mpqp/execution/runner.py

```diff
@@ -112,13 +112,12 @@ def generate_job(
     elif nb_meas == 1:
         measurement = m_list[0]
         if isinstance(measurement, BasisMeasure):
-            # TODO: handle other basis by adding the right rotation (change
-            # of basis) before measuring in the computational basis
-            # Muhammad: circuit.add(CustomGate(UnitaryMatrix(change_of_basis_inverse)))
+            modified_circuit = circuit.without_measurements() + measurement.pre_measure
```

Comment:
According to the discussion we had last time, we said that it would be more relevant to let the pre_measure handling at the provider part right?
In the sense that when the user wants to retrieve the circuit from the job (in the local case), the pre_measure doesn't appear. We also said that it can impact noisy simulations (where we expect that only the circuit gates are affected by noise, not the pre-measure we add for basis change that is just an ergonomic feature).

In any case, if we choose to do it here, we have to remove the specific treatment we do for some providers (for example line 144 in google.py, this can be the source of the problems we face with Cirq i think).

Reply:
hmm, good point, I think we are going to change the semantics of `job.circuit` because the alternative is extremely painful (and error prone)

Reply:
Done in `mpqp-to-qasm` branch, waiting for the merge in this branch.